public_html/new_html/src/WikiText/WikiParse/src/DataModel/Template.php

```
@@ -0,0 +1,99 @@
+
+    public function changeParametersNames(array $params_new): void
+    {
+        $newParameters = [];
+        foreach ($this->parameters as $k => $v) {
+            $k = isset($params_new[$k]) ? $params_new[$k] : $k;
+            $newParameters[$k] = $v;
+        }
+        $this->parameters = $newParameters;
+    }
+
+    public function toString(bool $newLine = false, $ljust = 0): string
```

PHPMD (2.15.0)

The method toString has a boolean flag argument $newLine, which is a certain sign of a Single Responsibility Principle violation. (Clean Code Rules)

(BooleanArgumentFlag)